monoDriveIO/monodrive-client
# Diagnostics sensor tests + usability improvements

## Overview
This PR includes client facing diagnostic tests for along with some general hardening/usability improvements

## Changes
- adds fps diagnostic for camera, radar, lidar, ultrasonic, IMU, and state sensor
- adds and uses a new get license command to verify server license status before running
- implements simple logging system for cleaner output and more obvious errors
- throw exception on failed config file load

## Testing
- run all diagnostic test with `ctest --verbose` from your `./build` directory
- test that the client fails gracefully when the server is unlicensed
- general evaluate logging approach (open to other suggestions or patterns)
- make sure that examples fail loudly then the file path or working directory is incorrect
